tony-core/src/main/java/com/linkedin/tony/ReusablePort.java

```diff
@@ -44,54 +44,84 @@
  */
 final class ReusablePort extends ServerPort {
   private static final Log LOG = LogFactory.getLog(ReusablePort.class);
-  final EventLoopGroup eventLoopGroup;
-  final ChannelFuture future;
+  private final Process socketProcess;
+  private final int port;
+  public static final Path RESERVE_PORT_SCRIPT_PATH = requireNonNull(createPortReserveScript());
+  public static final String PORT_FILE_NAME_SUFFIX = "___PORT___";
+
+  private static Path createPortReserveScript() {
+    ClassLoader classloader = Thread.currentThread().getContextClassLoader();
+    final String reservePortScript = "reserve_reusable_port.py";
+    try {
+      // copy reserve_reusable_port.py from resource dir to a tmp dir
+      Path tempDir = Files.createTempDirectory("reserve_reusable_port");
+      tempDir.toFile().deleteOnExit();
+      try (InputStream stream = classloader.getResourceAsStream(reservePortScript)) {
+        Files.copy(stream, Paths.get(tempDir.toAbsolutePath().toString(), reservePortScript));
+      }
+      return Paths.get(tempDir.toAbsolutePath().toString(), reservePortScript);
+    } catch (IOException ex) {
+      return null;
```

Comment:
what does this mean? Is it expected to have exception here?

Reply:
when exception is thrown, just return an invalid path.